### PR TITLE
feat: add service endpoint for failed bull events

### PR DESCRIPTION
**Pull request by maxdeichmann**
> [!IMPORTANT]
> Adds `EvalExecutionQueue` for managing evaluation jobs and an API endpoint for retrying failed jobs, integrating it into the system.
> 
>   - **Behavior**:
>     - Adds `EvalExecutionQueue` class in `evalExecutionQueue.ts` for managing evaluation jobs with retry logic and error handling.
>     - Introduces API endpoint in `index.ts` to retry failed BullMQ jobs, restricted to Langfuse Cloud.
>   - **Integration**:
>     - Integrates `EvalExecutionQueue` into `evalService.ts` for creating and managing evaluation jobs.
>     - Updates `workerManager.ts` to register and manage workers for `EvalExecutionQueue`.
>     - Modifies `evalQueue.ts` to use `EvalExecutionQueue` for processing evaluation jobs.
>   - **Misc**:
>     - Removes `EvalExecutionQueue` definition from `evalQueue.ts` and imports it from shared server.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=langfuse%2Flangfuse&utm_source=github&utm_medium=referral)<sup> for 0b8b9b87cfae95de92126694e520f160d6d67dfa. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: web/src/pages/api/admin/bullmq/index.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+            logger.warn(
+              `Circuit breaker activated: Stopped after ${maxLoops} iterations for queue ${queueName}`,
+            );
+            break;
+          }
+
+          failed = await queue?.getJobs(["failed"], 0, 100, true);
+          logger.info(`Retrying jobs for queue ${JSON.stringify(failed)}`);
+          if (failed && failed.length > 0) {
+            await Promise.all(failed.map((job) => job.retry()));
```
Comment (maxdeichmann) [inline web/src/pages/api/admin/bullmq/index.ts:92]: the retry() function seems to do the trick. At least this is what i tested on local dev.

File: web/src/pages/api/admin/bullmq/index.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        do {
+          if (loopCount >= maxLoops) {
+            logger.warn(
+              `Circuit breaker activated: Stopped after ${maxLoops} iterations for queue ${queueName}`,
+            );
+            break;
+          }
+
+          failed = await queue?.getJobs(["failed"], 0, 100, true);
+          logger.info(`Retrying jobs for queue ${JSON.stringify(failed)}`);
```
Comment (Steffen911) [inline web/src/pages/api/admin/bullmq/index.ts:90]: What is in the `failed` object? Can it contain PII? In that case we should rather log the count or the queue name.

If it's just the job config, good to go.

**Review by Steffen911 — APPROVED** [review]

File: web/src/pages/api/admin/bullmq/index.ts (hunk 2 above)
Comment (maxdeichmann) [reply web/src/pages/api/admin/bullmq/index.ts:90]: true. left over from testing. Removed.

File: web/src/pages/api/admin/bullmq/index.ts (hunk 1 above)
Comment (Steffen911) [reply web/src/pages/api/admin/bullmq/index.ts:91]: Looking good based on the bullmq api declaration as well.

**Pull request merged**